### PR TITLE
proto: modify height int64 to uint64

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- change `height`, `base` from int64 to uint64

### CODE REVIEW

**Comment by cmwaters** [comment]
I think `proposer_priority` needs to remain a int64. I believe it goes negative after a validator has just proposed. I'm sure we could offset by a value so that all proposer priorities are non negative numbers but I think this will involve tweaking the algorithm.

Also do you plan to do other number types i.e for `round` int32 -> uint32 in a separate PR

By `base`, do you mean `initial_height`?

**Comment by marbar3778** [comment]
> I think `proposer_priority` needs to remain a int64. I believe it goes negative after a validator has just proposed. I'm sure we could offset by a value so that all proposer priorities are non negative numbers but I think this will involve tweaking the algorithm.

Was just looking into tendermint to get a better understanding if its possible. Thanks for the comment, saved me some time.

File: proto/abci/types.proto [hunk 1]
```diff
@@ -322,13 +322,13 @@ message TxResult {
 message Validator {
   bytes address = 1;  // The first 20 bytes of SHA256(public key)
   // PubKey pub_key = 2 [(gogoproto.nullable)=false];
-  int64 power = 3;  // The voting power
+  uint64 power = 3;  // The voting power
```
Comment (marbar3778) [inline proto/abci/types.proto:325]:
```suggestion
  int64 power = 3;  // The voting power
```

**Comment by marbar3778** [comment]
making this pr only contain changes to height

File: proto/abci/types.proto [hunk 2]
```diff
@@ -90,7 +90,7 @@ message RequestDeliverTx {
 }
 
 message RequestEndBlock {
-  int64 height = 1;
```
Comment (cmwaters) [inline proto/abci/types.proto:93]: This is a bit of an aside and perhaps not relevant to this PR, but don't you think that this file should have comments on what exactly the data structures and methods are for? Or is this covered elsewhere in the spec

**Review by cmwaters — APPROVED** [review]
Thumbs up from me 👍 

You just missed `LightClientAttackEvidence.CommonHeight` as far as I can see

**Review by melekes — APPROVED** [review]
👍

File: proto/abci/types.proto (hunk 2 above)
Comment (marbar3778) [reply proto/abci/types.proto:93]: The displayed docs are located here: https://docs.tendermint.com/master/spec/p2p/messages/. We can add comments here, they will add value to us and tm-rs team

**Pull request merged**